pkg/kubectl/cmd/describe.go

```diff
@@ -88,3 +95,39 @@ func RunDescribe(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []s
 	fmt.Fprintf(out, "%s\n", s)
 	return nil
 }
+
+func DescribeMatchingResources(mapper meta.RESTMapper, typer runtime.ObjectTyper, describer kubectl.Describer, f *cmdutil.Factory, namespace, rsrc, prefix string, out io.Writer) error {
+	r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+		NamespaceParam(namespace).DefaultNamespace().
+		ResourceTypeOrNameArgs(true, rsrc).
+		SingleResourceType().
+		Do()
```

Comment:
Would be nice if you could reuse the builder (to guard against bit rot) but it's not easy to do here I don't think.